src/org/traccar/geocoder/Geocoder.java

```diff
@@ -20,11 +20,13 @@
     interface ReverseGeocoderCallback {
 
         void onSuccess(String address);
+        void onSuccess(double speed);
 
         void onFailure(Throwable e);
 
     }
 
     String getAddress(double latitude, double longitude, ReverseGeocoderCallback callback);
+    double getSpeedLimit(double latitude, double longitude, ReverseGeocoderCallback callback);
```

Comment:
getAddress returns a string which is accessed by several other files as mentioned in previous comment. I like the idea of attaching speed limit to position. The other way, which is more caching-friendly is to store lat,lon against a place_id and store the speed limit against place_id. This will reduce netowrk calls overall but will require quite a few more changes to implement, plus changes to the way other files call the geocoder.